repo/packages/S/spark/13/config.json

```diff
@@ -0,0 +1,235 @@
+{
+    "type": "object",
+    "properties": {
+        "service": {
+            "description": "DCOS Spark configuration properties",
+            "type": "object",
+            "properties": {
+                "name": {
+                    "default": "spark",
+                    "description": "The Spark Dispatcher will register with Mesos with this as a framework name.  This service will be available at http://<dcos_url>/service/<name>/",
+                    "type": "string"
+                },
+                "cpus": {
+                    "default": 1,
+                    "description": "CPU shares",
+                    "minimum": 0.0,
+                    "type": "number"
+                },
+                "mem": {
+                    "default": 1024.0,
+                    "description": "Memory (MB)",
+                    "minimum": 1024.0,
+                    "type": "number"
+                },
+                "role": {
+                    "description": "The Spark Dispatcher will register with Mesos with this role.",
+                    "type": "string",
+                    "default": "*"
+                },
+                "principal": {
+                    "description": "The Spark Dispatcher will register with Mesos with this principal.",
+                    "type": "string",
+                    "default": ""
+                },
+                "secret": {
+                    "description": "The Spark Dispatcher will register with mesos with this secret.",
+                    "type": "string",
+                    "default": ""
+                },
```

Comment:
Ok, that's what I thought. It's for generic framework authentication, but the `credential.secret` is only used for CRAM_MD5. We definitely want the service account principal in MesosSchedulerDriver(...,credential), and our authenticatee just ignores the contents of the credential secret. Instead we read the contents of DCOS_SERVICE_ACCOUNT_CREDENTIAL, which is populated by the secrets hook after the user specifies this field.
Currently, you can only set MESOS_AUTHENTICATEE to "com_mesosphere_dcos_ClassicRPCAuthenticatee" or "cram_md5" (or our old, not-so-supported kerberos authn), so if you're setting MESOS_AUTHENTICATEE, then you're not using Credential.secret. But I guess you can't have the condition go both ways.. circular logic.

Separate field or a checkbox would be the solution then. Not sure how much sense it really makes to reuse this field. Maybe @nickbp has some input based on how the rest of Infinity plans to do it.